node/src/actors/chain_manager/mod.rs

```diff
@@ -970,7 +970,25 @@ impl ChainManager {
         })
         .map_err(|e, _, _| log::error!("Superblock building failed: {:?}", e))
         .and_then(move |(block_headers, last_hash), act, _ctx| {
-            let ars_members = &act.chain_state.last_ars;
+            let chain_info = act.chain_state.chain_info.as_ref().unwrap();
+            let x;
+            let ars_members = {
+                // Before reaching the epoch activity_period + collateral_age the bootstrap committee signs the superblock
+                if act.current_epoch.unwrap()
+                    > chain_info.consensus_constants.collateral_age
+                        + chain_info.consensus_constants.activity_period
```

Comment:
`collateral_age` is measured in blocks, while `activity_period` is measured in epochs. Are we OK to add these together?

Reply:
The idea is to leave decentralized when we are sure that the ARS doesnt contain shitty nodes. So the idea was to wait until requests are living for 1 activity period (stressed net). But as Adan said, if there are empty epochs, the collateral age is not the same, because it is counted in blocks, so should we use an offset?

Reply:
I see your concern with a small ARS by the time this timeframe elapses. But was not it the reason why we decided to put this timeframe in the first place? Perhaps we can double it if we feel safer